File: src/main/java/client/messages/quest/CompleteQuestRecordMessage.java
package client.messages.quest;

import client.player.quest.QuestState;
import util.packet.PacketWriter;

public class CompleteQuestRecordMessage extends AbstractQuestRecordMessage {

    private final long dateCompleted;

    public CompleteQuestRecordMessage(short questId, long dateCompleted) {
        super(questId);
        this.dateCompleted = dateCompleted;
    }

    @Override
    public QuestState getState() {
        return QuestState.COMPLETE;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        super.encodeData(pw);
        pw.writeLong(dateCompleted);
    }
}

File: src/main/java/client/messages/quest/CompleteQuestRecordMessage.kt
package client.messages.quest

import client.player.quest.QuestState
import util.packet.PacketWriter

class CompleteQuestRecordMessage(questId: Short, private val dateCompleted: Long) : AbstractQuestRecordMessage(questId) {

    override val state: QuestState get() = QuestState.COMPLETE

    override fun encodeData(pw: PacketWriter) {
        super.encodeData(pw)
        pw.writeLong(dateCompleted)
    }
}